docs/concepts/models.md

```diff
@@ -66,77 +74,96 @@ In this example, `User` is a model with two fields:
 * `id`, which is an integer and is required
 * `name`, which is a string and is not required (it has a default value).
 
+The model can then be instantiated:
+
 ```py group="basic-model"
 user = User(id='123')
 ```
 
-In this example, `user` is an instance of `User`.
-Initialization of the object will perform all parsing and validation.
-If no `ValidationError` is raised, you know the resulting model instance is valid.
+`user` is an instance of `User`. Initialization of the object will perform all parsing and validation.
+If no [`ValidationError`][pydantic_core.ValidationError] exception is raised,
+you know the resulting model instance is valid.
+
+Fields of a model can be accessed as normal attributes of the `user` object:
 
 ```py group="basic-model"
+assert user.name == 'Jane Doe'  # (1)!
 assert user.id == 123
 assert isinstance(user.id, int)
-# Note that '123' was coerced to an int and its value is 123
 ```
 
-More details on pydantic's coercion logic can be found in [Data Conversion](#data-conversion).
-Fields of a model can be accessed as normal attributes of the `user` object.
-The string `'123'` has been converted into an int as per the field type.
+1. `name` wasn't set when `user` was initialized, so the default value was used.
```

Comment:
The following changes are mainly moving example snippets below the text describing it.